### PR TITLE
finish problems through problem 3

### PR DESCRIPTION
**Pull request by rdripley**
@paulstatezny tried to do problem 4. The part I'm struggling with is the function portion. I tried what I thought would be the easiest option and made my parent and child classes only contain a public variable that I would then call using the function to print out a string. That did not work :( 
I keep getting an error-> Fatal error: Call to undefined method GreatMovie::pickMovie()

### CODE REVIEW

**Comment by rdripley** [comment]
@paulstatezny I tried my hand at Problem 7. The error I run into is "Missing argument" but I'm not sure what argument to put in to fix that error message.

**Comment by rdripley** [comment]
@paulstatezny I got the code to do everything I want. I just have 3 questions that I will also probably ask for clarification when we talk tomorrow.
1. Why do I need the function printMessage after creating the interface? Or do I even need it? Is there another way to do it?
2. When I tried to call function printMessage the first time, I didn't have both public functions. 
   I had public function print($message) and that was it. Why won't it let me use print()? It gives me an error every time I try. I replaced it with work($message) and it worked a lot better.
3. Why do you need to have 2 functions? I couldn't get it to work with just one.

File: php-practice/section13.php [hunk 1]
```diff
@@ -56,6 +56,12 @@ class TwelveSidedDie extends Dice {
  * Let the PREFIX be "Error: " and specify it in a class constant called PREFIX.
  */
 
+class ErrorOutPutter {
+  const PREFIX = "Error: ";
+  public function outputError($error) {
+    echo self::PREFIX . $error;
+  }
+}
```
Comment (paulstatezny) [inline php-practice/section13.php:64]: Great job!

File: php-practice/section13.php [hunk 2]
```diff
@@ -67,6 +73,11 @@ class TwelveSidedDie extends Dice {
  * Echo out the PREFIX from your ErrorOutputter class
  */
 
+$errorOutputter = new ErrorOutputter();
+
+$errorOutputter->outputError('Syntax Error');
+
+echo SEPARATOR;
```
Comment (paulstatezny) [inline php-practice/section13.php:80]: Actually the intention for this problem was to have you echo out the `PREFIX` constant from your `ErrorOutputter` class.

e.g.

``` php
echo TwelveSidedDie::SIDES; // Echos out 12 to the screen
```

File: php-practice/section13.php [hunk 3]
```diff
@@ -144,6 +156,23 @@ function takeDice(Dice $ball) {
...
+
+function pickMovie($greatMovie, $bestMovie) {
+  echo $greatMovie . 'isn\'t as good as' . $bestMovie;
+}
+
+$movieAnswer = new GreatMovie();
+
+/*$movieAnswer->pickMovie();*/
+
+echo SEPARATOR;
```
Comment (paulstatezny) [inline php-practice/section13.php:176]: I forgot to mention with this problem -- I'd like the `pickMovie` function to use _Typehinting_ to enforce that the correct types of arguments are provided to the function. (Just like the `takeDice` function above.)

File: php-practice/section13.php (hunk 2 above)
Comment (rdripley) [inline php-practice/section13.php:80]: Fixed

File: php-practice/section13.php [hunk 4]
```diff
@@ -224,3 +254,26 @@ function growPlant(Plant $plant) {
...
+    echo "Error: " . $message . PHP_EOL;
+  }
+}
+
+$notificationPrinter = new NotificationPrinter();
+$errorPrinter = new ErrorPrinter();
+
+$notificationPrinter->printMessage('Issue with syntax');
+
+$errorPrinter->printMessage('You forgot the semicolon, ya dummy!');
```
Comment (paulstatezny) [inline php-practice/section13.php:279]: haha

**Pull request merged**